src/libtego/include/tego/tego.h

```diff
@@ -826,6 +876,68 @@ void tego_context_send_message(
     tego_message_id_t* out_id,
     tego_error_t** error);
 
+/*
+ * Request to send a file to the given user
+ *
+ * @param context : the current tego context
+ * @param user : the user to send a file to
+ * @param filePath : utf8 path to file to send
+ * @param filePathLength : length of filePath not including null-terminator
+ * @param out_id : optional, filled with assigned file transfer id for callbacks
+ * @param out_fileHash : optional, filled with hash of the file to send
+ * @param out_fileSize : optional, filled with the size of the file in bytes
+ * @param error : filled on error
+ */
+void tego_context_send_file_transfer_request(
+    tego_context_t* context,
+    tego_user_id_t const*  user,
+    char const* filePath,
+    size_t filePathLength,
+    tego_file_transfer_id_t* out_id,
+    tego_file_hash_t** out_fileHash,
+    tego_file_size_t* out_fileSize,
+    tego_error_t** error);
+
+typedef enum
+{
+    tego_file_transfer_response_accept, // proceed with a file transfer
+    tego_file_transfer_response_reject, // reject the file transfer
+} tego_file_transfer_response_t;
```

Comment:
Hmm, not sure I agree. The response is an enum rather than a tego_bool_t in case we ended up needing different responses in the future (tego_file_transfer_response_maybe_later, tego_file_transfer_response_never_stop_asking, etc). I don't think it really matters, especially since this enum never actually makes its way down to the protofbuf layer.